### ISSUE TITLE
ASP.NET Core does have a SynchronizationContext when using Blazor

### ISSUE THREAD

**Issue by Sebazzz** [davidfowl/AspNetCoreDiagnosticScenarios#99]
[Avoid using Task.Result and Task.Wait](https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#avoid-using-taskresult-and-taskwait):

https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/0c82fa823796c38d0436c0c08665c944ab12f91d/AsyncGuidance.md?plain=1#L232

ASP.NET Core Blazor does appear to use a synchronization context, being the [RendererSynchronizationContext](https://source.dot.net/Microsoft.AspNetCore.Components/R/20c5f0a262ea5879.html)](https://source.dot.net/#Microsoft.AspNetCore.Components/Rendering/RendererSynchronizationContext.cs,11).

**Comment by davidfowl** [comment]
It does! Specifically, in Blazor Server when executing razor components (I should probably change that now).

**Comment by Sebazzz** [comment]
Yup, I should have specifically mentioned Blazor Server. This SynchronizationContext implementation doesn't necessarily schedule the work on the same thread like the WPF, WinForms and legacy ASP.NET does. It is prone to the same blocking issues however, I think.
